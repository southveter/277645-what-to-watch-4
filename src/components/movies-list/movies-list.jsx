import React, {PureComponent} from "react";
import PropTypes from "prop-types";
import {MovieCard} from '../movie-card/movie-card.jsx';

export class MoviesList extends PureComponent {
  constructor(props) {
    super(props);

    this.state = {
      activeCard: ``,
    };
  }


  render() {
    const {movies, onClick} = this.props;
    return (
      <div className="catalog__movies-list">
        {movies.map((movie) => (
          <MovieCard
            key={movie.title}
            movieInfo={movie}
            onClick={(evt) => {
              evt.preventDefault();
              onClick(movie);
            }}
            onHover={() => {
              this.setState({
                activeCard: movie
              });
            }}/>
        ))}
      </div>
    );
  }
}


MoviesList.propTypes = {
  movies: PropTypes.arrayOf(PropTypes.shape({
    poster: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
  })).isRequired,
  onClick: PropTypes.func.isRequired,
};
