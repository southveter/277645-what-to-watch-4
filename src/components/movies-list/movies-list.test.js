import React from "react";
import renderer from "react-test-renderer";
import {MoviesList} from './movies-list.jsx';


const mocks = [
  {
    title: `Fantastic Beasts: The Crimes of Grindelwald`,
    poster: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    source: `video/preview.webm`,
  },
  {
    title: `Pulp Fiction`,
    poster: `img/pulp-fiction.jpg`,
    source: `video/preview.webm`,
  },
  {
    title: `Shutter Island`,
    poster: `img/shutter-island.jpg`,
    source: `video/preview.webm`,
  },
  {
    title: `Macbeth`,
    poster: `img/macbeth.jpg`,
    source: `video/preview.webm`,
  },
  {
    title: `Snatch`,
    poster: `img/snatch.jpg`,
    source: `video/preview.webm`,
  },
  {
    title: `We need to talk about Kevin`,
    poster: `img/we-need-to-talk-about-kevin.jpg`,
    source: `video/preview.webm`,
  },
  {
    title: `Bohemian Rhapsody`,
    poster: `img/bohemian-rhapsody.jpg`,
    source: `video/preview.webm`,
  },
  {
    title: `Revenant`,
    poster: `img/revenant.jpg`,
    source: `video/preview.webm`,
  },
];
const onClick = () => {};

it(`Should render correctly`, () => {
  const tree = renderer
    .create(
        <MoviesList
          movies={mocks}
          onClick={onClick}
        />
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
