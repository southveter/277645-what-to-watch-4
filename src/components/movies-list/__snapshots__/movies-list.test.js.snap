// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render correctly 1`] = `
<div
  className="catalog__movies-list"
>
  <article
    className="small-movie-card catalog__movies-card"
  >
    <div
      className="small-movie-card__image"
      onMouseOver={[Function]}
    >
      <img
        alt="Fantastic Beasts: The Crimes of Grindelwald"
        height="175"
        onClick={[Function]}
        src="img/fantastic-beasts-the-crimes-of-grindelwald.jpg"
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
      onClick={[Function]}
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
      >
        Fantastic Beasts: The Crimes of Grindelwald
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
  >
    <div
      className="small-movie-card__image"
      onMouseOver={[Function]}
    >
      <img
        alt="Pulp Fiction"
        height="175"
        onClick={[Function]}
        src="img/pulp-fiction.jpg"
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
      onClick={[Function]}
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
      >
        Pulp Fiction
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
  >
    <div
      className="small-movie-card__image"
      onMouseOver={[Function]}
    >
      <img
        alt="Shutter Island"
        height="175"
        onClick={[Function]}
        src="img/shutter-island.jpg"
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
      onClick={[Function]}
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
      >
        Shutter Island
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
  >
    <div
      className="small-movie-card__image"
      onMouseOver={[Function]}
    >
      <img
        alt="Macbeth"
        height="175"
        onClick={[Function]}
        src="img/macbeth.jpg"
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
      onClick={[Function]}
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
      >
        Macbeth
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
  >
    <div
      className="small-movie-card__image"
      onMouseOver={[Function]}
    >
      <img
        alt="Snatch"
        height="175"
        onClick={[Function]}
        src="img/snatch.jpg"
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
      onClick={[Function]}
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
      >
        Snatch
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
  >
    <div
      className="small-movie-card__image"
      onMouseOver={[Function]}
    >
      <img
        alt="We need to talk about Kevin"
        height="175"
        onClick={[Function]}
        src="img/we-need-to-talk-about-kevin.jpg"
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
      onClick={[Function]}
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
      >
        We need to talk about Kevin
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
  >
    <div
      className="small-movie-card__image"
      onMouseOver={[Function]}
    >
      <img
        alt="Bohemian Rhapsody"
        height="175"
        onClick={[Function]}
        src="img/bohemian-rhapsody.jpg"
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
      onClick={[Function]}
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
      >
        Bohemian Rhapsody
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
  >
    <div
      className="small-movie-card__image"
      onMouseOver={[Function]}
    >
      <img
        alt="Revenant"
        height="175"
        onClick={[Function]}
        src="img/revenant.jpg"
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
      onClick={[Function]}
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
      >
        Revenant
      </a>
    </h3>
  </article>
</div>
`;
